File: src/FsPickler/FsPickler/XmlSerializer.fs
namespace MBrace.FsPickler

/// <summary>
///     XML pickler instance.
/// </summary>
[<Sealed; AutoSerializable(false)>]
type XmlSerializer =
    inherit FsPicklerTextSerializer
        
    val private format : XmlPickleFormatProvider

    /// <summary>
    ///     Define a new Xml pickler instance.
    /// </summary>
    /// <param name="indent">Enable indentation of output XML pickles.</param>
    /// <param name="typeConverter">Define a custom type name converter.</param>
    /// <param name="picklerResolver">Specify a custom pickler resolver/cache for serialization. Defaults to the singleton pickler cache.</param>
    new ([<O;D(null)>] ?indent : bool, [<O;D(null)>] ?typeConverter : ITypeNameConverter, [<O;D(null)>]?picklerResolver : IPicklerResolver) =
        let xml = new XmlPickleFormatProvider(defaultArg indent false)
        { 
            inherit FsPicklerTextSerializer(xml, ?typeConverter = typeConverter, ?picklerResolver = picklerResolver)
            format = xml    
        }

    /// Gets or sets indentation of serialized pickles.
    member x.Indent
        with get () = x.format.Indent
        and set b = x.format.Indent <- b

File: src/FsPickler.CSharp/Formats/XmlSerializer.cs
using System;
using System.Reflection;
using System.IO;
using System.Text;
using System.Runtime.Serialization;

using FSP = MBrace.FsPickler;

namespace MBrace.CsPickler
{
    /// <summary>
    ///     Defines an Xml serializer instance.
    /// </summary>
    public class XmlSerializer : CsPicklerTextSerializer
    {
        private FSP.XmlSerializer _xs;

        private XmlSerializer(FSP.XmlSerializer xs)
            : base(xs) 
        {
            _xs = xs;
        }

        /// <summary>
        ///     Creates a new XmlPickler instance.
        /// </summary>
        /// <param name="indent">indent xml serializations; defaults to false.</param>
        public static XmlSerializer Create(bool indent = false)
        {
            var xs = new FSP.XmlSerializer();
            xs.Indent = indent;
            return new XmlSerializer(xs);
        }

        /// <summary>
        ///     Gets or sets indentation of Xml serializations.
        /// </summary>
        public bool Indent
        {
            get { return _xs.Indent ; }
            set { _xs.Indent = value; }
        }
    }
}
